catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -873,4 +888,25 @@ public String getSortKey(String field) {
   public Stream<String> anyTextFields() {
     return anyTextFieldsCache.stream();
   }
+
+  /**
+   * Gets the metadata size limit from custom.system.properties. Defaults to 5242880 bytes.
+   *
+   * @return Integer metadata size limit in bytes
+   */
+  private static Integer getMetadataSizeLimit() {
+    String x =
+        AccessController.doPrivileged(
+            (PrivilegedAction<String>)
+                () ->
+                    System.getProperty(
+                        METADATA_SIZE_LIMIT, String.valueOf(FIVE_MEGABYTES_BINARY_IN_BYTES)));
+    if (StringUtils.isNumeric(x)) {
+      int size = Integer.parseInt(x);
+      if (size < Integer.MAX_VALUE) {
```

Comment:
❓ I don't think size can ever be larger than `Integer.MAX_VALUE` unless you parse it as a long. It will overflow

Reply:
changed logic account for this